File: java/test/src/main/java/test/Ice/inheritance/IB2I.java
// Copyright (c) ZeroC, Inc.

package test.Ice.inheritance;

import com.zeroc.Ice.Current;

import test.Ice.inheritance.Test.MA.IAPrx;
import test.Ice.inheritance.Test.MB.IB2;
import test.Ice.inheritance.Test.MB.IB2Prx;

public final class IB2I implements IB2 {
    public IB2I() {}

    @Override
    public IAPrx iaop(IAPrx p, Current current) {
        return p;
    }

    @Override
    public IB2Prx ib2op(IB2Prx p, Current current) {
        return p;
    }
}

File: csharp/test/Ice/inheritance/IB2I.cs
// Copyright (c) ZeroC, Inc. All rights reserved.

using System.Threading;

namespace ZeroC.Ice.Test.Inheritance
{
    public sealed class B2 : MB.IB2
    {
        public MA.IAPrx? Iaop(MA.IAPrx? p, Current current, CancellationToken cancel) => p;

        public MB.IB2Prx? Ib2op(MB.IB2Prx? p, Current current, CancellationToken cancel) => p;
    }
}
